### PR TITLE
#70: Update app properties in readme

### PR DESCRIPTION
**Pull request by jozefbakus**
Implements #70

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,38 +77,59 @@ Adjusted application properties have to be provided. An application properties t
 version=@project.version@
 # Enviroment where application will be running
 environment=Local
+# Maximum number of workflows that can be trigger in bulk (Default value 10)
```
Comment (kevinwallimann) [inline README.md]: typo: trigger -> triggered

File: README.md [hunk 2]
```diff
@@ -77,38 +77,59 @@ Adjusted application properties have to be provided. An application properties t
...
+```
+# Health check settings
+health.databaseConnection.timeoutMillis=120000
+health.yarnConnection.testEndpoint=/cluster/cluster
+health.yarnConnection.timeoutMillis=60000
 ```
 ```
 # How will users authenticate. Available options: inmemory, ldap
 auth.mechanism=inmemory
+#If set all users that does not have admim role will not have access to admin protected endpoints
```
Comment (kevinwallimann) [inline README.md]: `If set all` -> `If set, all` (missing comma). `that does` -> `that do`

**Review by kevinwallimann — COMMENTED** [review]
Some keys have values, some don't. What's the pattern here? Not all keys that have default values, are written here.

E.g. spark.submit.thread.pool.size is 10 by default, but it's `spark.submit.thread.pool.size=` and not `spark.submit.thread.pool.size=10`

File: README.md (hunk 2 above)
Comment (jozefbakus) [reply README.md:96]: Done

File: README.md (hunk 1 above)
Comment (jozefbakus) [reply README.md:80]: Done

Comment (jozefbakus) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (jozefbakus) [reply README.md]: Fixed

**Comment by jozefbakus** [comment]
> Some keys have values, some don't. What's the pattern here? Not all keys that have default values, are written here.
> 
> E.g. spark.submit.thread.pool.size is 10 by default, but it's `spark.submit.thread.pool.size=` and not `spark.submit.thread.pool.size=10`

Missed that one, but I defined values for keys that you can guess. 
You know what to set for `notification.enabled=false`, but you no idea what to set for `notification.sender.address=`

File: README.md [hunk 3]
```diff
@@ -77,38 +77,59 @@ Adjusted application properties have to be provided. An application properties t
...
+notification.sender.address=
+spring.mail.host=
+spring.mail.port=
+```
+```
 #Kafka sensor properties. Not all are required. Adjust according to your use case.
-kafkaSource.group.id=hyper_drive_${appUniqueId}
+kafkaSource.group.id.prefix=hyper_drive_${appUniqueId}
 kafkaSource.key.deserializer=org.apache.kafka.common.serialization.StringDeserializer
 kafkaSource.value.deserializer=org.apache.kafka.common.serialization.StringDeserializer
```
Comment (kevinwallimann) [inline README.md]: This changed. Now it's `kafkaSource.properties.key.deserializer` and `kafkaSource.properties.value.deserializer`. Also `kafkaSource.properties.max.poll.records`

File: README.md [hunk 4]
```diff
@@ -77,38 +77,59 @@ Adjusted application properties have to be provided. An application properties t
...
+# Maximum number of workflows that can be triggered in bulk (Default value 10)
+application.maximumNumberOfWorkflowsInBulkRun=10
+```
+```
+# Unique app id, for easier application identification
+appUniqueId=
+```
+```
+# Health check settings
+health.databaseConnection.timeoutMillis=120000
```
Comment (kevinwallimann) [inline README.md:89]: Could you change the corresponding default value in `application.properties`?

File: README.md [hunk 5]
```diff
@@ -77,38 +77,59 @@ Adjusted application properties have to be provided. An application properties t
...
+health.yarnConnection.testEndpoint=/cluster/cluster
+health.yarnConnection.timeoutMillis=60000
 ```
 ```
 # How will users authenticate. Available options: inmemory, ldap
 auth.mechanism=inmemory
+#If set, all users that do not have admim role will not have access to admin protected endpoints
+auth.admin.role=
```
Comment (kevinwallimann) [inline README.md]: In `application.properties`, the value is `ROLE_ADMIN`, but that's just a suggestion, there is no default value in the code. So, what do we do here?

File: README.md [hunk 6]
```diff
@@ -77,38 +77,59 @@ Adjusted application properties have to be provided. An application properties t
...
 # INMEMORY authentication: username and password defined here will be used for authentication.
 auth.inmemory.user=user
 auth.inmemory.password=password
+auth.inmemory.admin.user=admin-user
+auth.inmemory.admin.password=admin-passowrd
```
Comment (kevinwallimann) [inline README.md]: The default value is `hyperdriver-admin-user` and `hyperdriver-admin-password`. I'd change it to that, and also in the `application.properties`

File: README.md [hunk 7]
```diff
@@ -126,13 +147,20 @@ kafkaSource.properties.sasl.mechanism=
 kafkaSource.properties.sasl.jaas.config=
 ```
 ```
-#Spark yarn sink properties. Properties used to deploy and run Spark job in Yarn. Not all are required. Adjust according to your use case.
-spark.submit.thread.pool.size=
-sparkYarnSink.hadoopResourceManagerUrlBase=
+# Recurring sensor properties.
+recurring-sensor.maxJobsPerDuration=8
+recurring-sensor.duration=1h
```
Comment (kevinwallimann) [inline README.md]: I would change this to `recurringSensor` for consistency with `kafkaSource`. In the prefix annotation in `RecurringSensorConfig`, I have to use kebab case (as described here: https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#features.external-config.typesafe-configuration-properties.relaxed-binding), but camel case can be used in the application property

File: README.md [hunk 8]
```diff
@@ -126,13 +147,20 @@ kafkaSource.properties.sasl.mechanism=
...
+```
+```
+#Spark properties. Properties used to deploy and run Spark job. Not all are required. Adjust according to your use case.
+#Where spark jobs will be executed. Available options: yarn, emr.
+spark.submitApi=
```
Comment (kevinwallimann) [inline README.md]: Default value is `yarn`

File: README.md [hunk 9]
```diff
@@ -126,13 +147,20 @@ kafkaSource.properties.sasl.mechanism=
...
+
+#Submit api = YARN
+sparkYarnSink.submitTimeout=160000
 sparkYarnSink.hadoopConfDir=
 sparkYarnSink.sparkHome=
```
Comment (kevinwallimann) [inline README.md]: In `application.properties`, we give a value (probably a suggestion) for `sparkYarnSink.hadoopConfDir` and `sparkYarnSink.sparkHome` (`/opt/hadoop` and `/opt/spark`). Should we also add it here, or remove it in `application.properties` or we deliberately keep a difference?

File: README.md [hunk 10]
```diff
@@ -145,6 +173,16 @@ sparkYarnSink.additionalConfs.spark.yarn.keytab=
...
+spark.emr.filesToDeploy=
+spark.emr.additionalConfs=
+
+#Common properties for Submit api = YARN and EMR
+sparkYarnSink.hadoopResourceManagerUrlBase=
+sparkYarnSink.userUsedToKillJob=
+spark.submit.thread.pool.size=10
 ```
 ```
 #Postgresql properties for connection to trigger database
```
Comment (kevinwallimann) [inline README.md:188]: I would also add `spring.liquibase.change-log=classpath:/db_scripts/liquibase/db.changelog.yml`, otherwise the app doesn't work.

Comment (jozefbakus) [reply README.md:188]: Done

File: README.md (hunk 9 above)
Comment (jozefbakus) [reply README.md]: I added it into readme as well

File: README.md (hunk 7 above)
Comment (jozefbakus) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (jozefbakus) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (jozefbakus) [reply README.md:89]: Done

File: README.md (hunk 5 above)
Comment (jozefbakus) [reply README.md]: no value == no key == disabled, I added ROLE_ADMIN

File: README.md (hunk 6 above)
Comment (jozefbakus) [reply README.md]: Done

File: README.md (hunk 8 above)
Comment (jozefbakus) [reply README.md]: Done

**Review by kevinwallimann — APPROVED** [review]

**Pull request merged**